hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/sps/BlockStorageMovementNeeded.java

```diff
@@ -232,6 +233,7 @@ public synchronized void clearQueuesWithNotification() {
     public void run() {
       LOG.info("Starting SPSPathIdProcessor!.");
       Long startINode = null;
+      int retryCount = 0;
```

Comment:
Do we still need to implement the retry logic after catching the FileNotFoundException?

Reply:
@tasanuma Thanks for your reviews.
I'm not sure if there are any other exceptions may cause this thread work incorrectly.
And the retry logic may help to avoid this thread falling into an infinite loop if the 'startINode' fail to be set to null.

Reply:
@liubingxing Thanks for the explanation. I got it.

Reply:
Thanks @tasanuma